oak-store-spi/src/main/java/org/apache/jackrabbit/oak/plugins/memory/AbstractBlob.java

```diff
@@ -66,7 +66,7 @@ public static boolean equal(Blob a, Blob b) {
 
         //Check for identity first. If they are same then its
         //definitely same blob. If not we need to check further.
-        if (ai != null && bi != null && ai.equals(bi)){
+        if (ai != null && ai.equals(bi)){
             return true;
         }
 
```

Comment:
On that occasion, we could also simplify the length check to "if (a.length() != b.length())".